### PR TITLE
5 individual article page

### PR DESCRIPTION
**Pull request by ITedInnovator**

### CODE REVIEW

File: src/components/article-components/ArticleContent.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Link } from "react-router-dom";
+
+export const ArticleContent = ({title, imageURL, body, author, published, topic}) => {
+    const formattedDate = new Date(published);
+    console.dir(formattedDate);
```
Comment (dappernerddesigns) [inline src/components/article-components/ArticleContent.jsx]: Make sure to take these out.

File: src/pages/SingleArticle.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { ErrorComponent } from "../components/ErrorComponent";
+
+export const SingleArticle = () => {
+    const [ articleData, setArticleData ] = useState({});
+    const { article_id } = useParams();
+    const [ loading, setLoading ] = useState(true);
+    const [ error, setError ] = useState("");
+    
+    useEffect( () => {
+        getArticleById(setArticleData, setLoading, article_id).catch(err => {
```
Comment (dappernerddesigns) [inline src/pages/SingleArticle.jsx]: why is the set loading and setArticleData being sent to the api request?

File: src/api.js [hunk 3]
```diff
@@ -14,4 +14,15 @@ export const getAllArticles = (setArticles, setLoading) => {
         }
 
         )
+}
+
+export  const getArticleById = (setArticle, setLoading, article_id) => {
+    return instance.get(`/articles/${article_id}`).then(({data}) => {
+        setArticle(( currArticle ) => {
+            const articleData = data.article;
+
```
Comment (dappernerddesigns) [inline src/api.js]: I would expect to see these handled in the then chain of the useEffect rather than here.

**Review by dappernerddesigns — APPROVED** [review]
Nicely laid out. Remember to remove logs and have a look at my comments regarding setting state. Currently this works, only because this api request is being used in an useEffect. If you were to make the same call on a different component this would break.

**Review by cjplatten — COMMENTED** [review]
Looks good!

**Pull request merged**